src/Tmds.MDns/NetworkInterfaceHandler.cs

```diff
@@ -177,49 +181,60 @@ internal void OnServiceQuery(Name serviceName)
             }
         }
 
-        private void StartReceive()
+
+        private IPAddress GetSubnetMask()
+        {
+            return NetworkInterface.GetIPProperties().UnicastAddresses
+                                   .Where(a => a.Address.AddressFamily == AddressFamily.InterNetwork)
+                                   .Select(a => a.IPv4Mask)
+                                   .FirstOrDefault();
+        }
+
+        private IPAddress GetIPv4Address()
         {
-#if NETSTANDARD1_3
-            bool pending = _socket.ReceiveAsync(_receiveEventArgs);
-            if (!pending)
+            return NetworkInterface.GetIPProperties().UnicastAddresses
+                                   .Where(a => a.Address.AddressFamily == AddressFamily.InterNetwork)
+                                   .Select(a => a.Address)
+                                   .FirstOrDefault();
+        }
+
+        private bool IsUnreachable(IPAddress receivedFrom)
```

Comment:
Instead of this, can we use `SO_BINDTODEVICE`?

Reply:
I will have a look at this. Can this be used for both Linux and Windows using the SetRawSocketOption?

Reply:
Stepping back for a moment.

`IsUnreachable` filters the packets to the interface they are received on.
I would expect the socket to not receive packets for an interface it didn't request membership for.
So this is a false assumption?
Is it not true on Windows, on Linux, or both?

Reply:
It is false on Linux.
It works as long as the message was initiated from the service. So send a requests gets a reply on the same port.
But when a remote device reboots the message is received on all sockets. It will show up all on interfaces.

It is important for me as I want to only discover devices on a internal test network (IPv6 local link) and not the devices on the LAN.

Reply:
I have setup a new VM to test the behavior on Windows.
I made the IsUnreachable return false and on Windows the behavior is ok.
On Linux the ip addresses of the remote are received on both local adapters.
So, for example the IPv6%2 is not reachable as it is actually on %3. Also, as the addresses overwrite each other I can see
the ip addresses constantly change '~'